### ISSUE TITLE
Missing example of initializing class-typed (compound-valued) properties

### ISSUE THREAD

**Issue by fitdev** [dotnet/docs#40369]
### Type of issue

Missing information

### Description

Given 2 classes like this:

```cs
class A {
 
    public int I {get; set;} = 3;
    public bool B {get; set;} = true;
    public string S {get; set;} = "abc";
    public B ClassB {get; set;} = new() { BB = true, BI = 43 };
    public override string ToString() => $"{I}|{B}|{S}|||{ClassB}";
}

class B {
 
    public int BI {get; set;} = 23;
    public bool BB {get; set;} = false;
    public string BS {get; set;} = "BBBabc";
    public override string ToString() => $"{BI}|{BB}|{BS}";
}
```

The following example is missing:

```cs
var a = new A{
    I = 103,
    B = false,
    ClassB = { BI = 100003 } //Initializer, reuses current instance: ClassB's values will be: 100003 (new value we assign here), true (kept from A's initialization), "BBBabc" (unchanged default from B)
};
```

See example at [SharpLab](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgAwFgBQiCM70DcCGYABPkQLxFQCmA7kQIIDe6RrRAkuUZkgMwA0LNgCEuAM3wAbAM5VBaNkQDCk/NOmiKjIsM4UeSQ7yIBfIgHpz7KAEtgNqTYBeVMPyJgqEWUQDGEME8oYCIbKGlgfChfKnQTAG4cDEwATgAKfABKRLQ8QhIELmo6JiFWPW4+eUVNIgkZOTLlVXVa4rTMom1dLgMjUwtzADlaUPDI6Ni0BKS4VIyEbKT0RAYulia4YzCQzkYAcypgeKJZY7MKXhzFLaIAIwB7B8kdLsPj06P4i6JgSCprmxbnMkEQAMpvL6fc5cABE+DuvlhgNYt1EKjUGkhHzO3yKtA6XR0tT+EDkOgqABZjDMFEDjA9cK4wDYACZUIggogAFQeYL+YX2hLIAD4iAASWGMdgmGCMYSyxhg2UwOUY1omZFxWaFUTMBSbbbBCnYk64n4IK6G+5PF7CPXvM1fH71WQoznGLnCCEHKHmuH24QIpHu26M5lsjlc3n8llQIWdUUSqW6RX2tPKrXTIA===)

### Page URL

https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md

### Document Version Independent Id

95025282-20a0-ded8-c6d2-3b8ffff600c4

### Article author

@BillWagner

### Metadata

* ID: 0efd6a6a-b52a-f6a2-d98b-f245b55115ff 
* Service: **dotnet-csharp**
* Sub-service: **fundamentals**

**Comment by BillWagner** [comment]
Thanks for noting this @fitdev 

This is a useful enhancement for this article. I've added it to our backlog, and added the help wanted label as well.

**Comment by bigboybamo** [comment]
Hi @BillWagner I was wondering if I could pick this up? Where in the doc should the example be?

**Comment by BillWagner** [comment]
Hi @bigboybamo 

Thanks for picking this up!

I'd recommend adding the new samples before the [collection initializer](https://learn.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers#collection-initializers) section.

**Comment by bigboybamo** [comment]
Hi @BillWagner I have opened a pull request [here](https://github.com/dotnet/docs/pull/40466)

Also, I noticed there were some parts in the doc that are not consistent. 
Some sections use `System.Console.WriteLine` and some use `Console.WriteLine`
I don't know if it's something i can correct in this PR or in another?

**Comment by BartoszKlonowski** [comment]
@BillWagner I believe this issue was not linked to the PR before merging, hence this issue is still open. So as I was just passing by and noticed this, so letting you know just for your information that I think it can be closed.

**Comment by BillWagner** [comment]
Closed as fixed in #40466
